src/Microsoft.VisualStudio.Composition/Diagnostic/DiagnosticInfoCollector.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.VisualStudio.Composition.Diagnostic
+{
+    using System.Text;
+
+    internal class DiagnosticInfoCollector
+    {
+        private StringBuilder messages;
+
+        private DiagnosticInfoCollector()
+        {
+            this.messages = new StringBuilder();
+        }
+
+        public static DiagnosticInfoCollector CreateInstance()
+        {
+            return new DiagnosticInfoCollector();
+        }
+
+        public void Collect(string message)
+        {
+            this.messages.AppendLine(message);
+        }
+
+        public string GetAndClearInformation()
```

Comment:
Thanks for feedback !
Updated code to collect information only when Type is null and that would not allow to collect information for all type resolver. Also Used List to avoid allocation issue. Please provide your feedback if can be pushed. .